Assets/Scripts/UI/UIView.cs

```diff
@@ -43,6 +50,15 @@ public void EnableGameOverUI()
             quitButton.onClick.AddListener(OnQuitClicked);
         }
 
+        public void StoreHighScore()
+        {
+            if (currentScore > highScore)
+            {
+                highScore = currentScore;
+                PlayerPrefs.SetInt("HighScore", highScore);
+            }
+        }
+
```

Comment:
UI Service should not be the one responsible for saving data in Player Prefs, We will be abiding by Single Responsibility very strictly. We will create another Service for this functionality of needed, but not going to happen in UI.